src/QueryBuilder.php

```diff
@@ -113,6 +97,65 @@ public function getSearchParams(string $queryString, string $type, int $start, i
         return $params;
     }
 
+    /**
+     * @param string[] $blocks
+     */
+    private function buildConditions(string $index, array $blocks): array
+    {
+        $conditions = $this->buildConditionsByType($index, $blocks);
+
+        if ($this->includeAccessData($index)) {
+            $conditions[] = $this->buildAccessConditions();
+        }
+
+        return $conditions;
+    }
+
+    /**
+     * Use only simple input
+     * @param string[] $blocks
+     * @return string[]
+     */
+    private function getResultsConditions(array $blocks): array
+    {
+        $conditions = [];
+
+        foreach ($blocks as $block) {
+            $block  = $this->parseBlock($block);
+            if ($block->getField() === '') {
+                $conditions[] = sprintf('("%s")', $block->getTerm());
+            }
+        }
+
+        return $conditions;
+    }
+
+    private function buildConditionsByType(string $type, array $blocks): array
+    {
+        if ($type === IndexerInterface::DELIVERY_RESULTS_INDEX) {
+            return $this->getResultsConditions($blocks);
+        }
+
+        return $this->getResourceConditions($blocks);
+    }
+
+    private function getResourceConditions($blocks): array
+    {
+        foreach ($blocks as $block) {
+            $queryBlock = $this->parseBlock($block);
+
+            if (empty($queryBlock->getField())) {
+                $conditions[] = sprintf('("%s")', $queryBlock->getTerm());
+            } elseif ($this->isStandardField($queryBlock->getField())) {
+                $conditions[] = sprintf('(%s:"%s")', $queryBlock->getField(), $queryBlock->getTerm());
+            } else {
+                $conditions[] = $this->buildCustomConditions($queryBlock);
+            }
```

Comment:
Suggestion: I would extract this for readability and be prepared to extend future verifications.

```
private function getBlockCondition()
{
            if (empty($queryBlock->getField())) {
                return sprintf('("%s")', $queryBlock->getTerm());
            }

            if ($this->isStandardField($queryBlock->getField())) {
                return sprintf('(%s:"%s")', $queryBlock->getField(), $queryBlock->getTerm());
            }

            return $this->buildCustomConditions($queryBlock);
}
```

Reply:
I don't like this part of code. I separate it slightly so we can zoom out and decide how we could refactor this on design level. I consider this as bigger task that will take care of this poor design in the future

Reply:
I am asking, cause you touched and refactored, so I am not asking to change something unrelated to what you did, but this is up to you. :)